impl/src/main/java/org/jboss/shrinkwrap/impl/BeansXml.java

```diff
@@ -239,9 +251,19 @@ public BeansXml setBeanDiscoveryMode(BeanDiscoveryMode mode) {
     @Override
     public InputStream openStream() {
         StringBuilder xml = new StringBuilder();
-        xml.append("<beans version=\"" + getBeansXmlVersion().getValue() + "\" bean-discovery-mode=\"");
-        xml.append(getBeanDiscoveryMode().getValue());
-        xml.append("\">\n");
+        xml.append("<beans");
+        BeansXmlVersion version = getBeansXmlVersion();
+        // add version if required
+        if (version != null) {
+            xml.append(" version=\"" + getBeansXmlVersion().getValue() + "\"");
+        }
+        BeanDiscoveryMode mode = getBeanDiscoveryMode();
+        // add bean discovery mode if required
+        if (mode != null) {
+            xml.append(" bean-discovery-mode=\"" + getBeanDiscoveryMode().getValue() + "\"");
```

Comment:
Redundant `getBeanDiscoveryMode()` call:

```suggestion
            xml.append(" bean-discovery-mode=\"" + mode.getValue() + "\"");
```